

TebLocalPlannerROS:

 odom_topic: /carla/ego_vehicle/odometry
 map_frame: map
 controller_frequency: 15.0

 cmd_angle_instead_rotvel: False # Somehow False works better without this param even tho we have ackermann car
 wheelbase: 2.5

 oscillation_recovery: False
 #Viapoints
 global_plan_viapoint_sep: 5

 # Trajectory
  
 teb_autosize: True
 dt_ref: 0.3 # 0.3 incresed for performance
 dt_hysteresis: 0.2
 global_plan_overwrite_orientation: True
 max_global_plan_lookahead_dist: 50.0
 feasibility_check_no_poses: 5
    
 # Robot
         
 max_vel_x: 10 # 0.4
 max_vel_x_backwards: 4 # 0.2 
 max_vel_theta: 1.0 #0.3
 acc_lim_x: 3 #0.5
 acc_lim_theta: 1.0 #0.5
 min_turning_radius: 3
 footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
   type: "polygon"
   vertices: [[-2, -1], [-2, 1], [2, 1], [2, -1]] # for type "polygon"

 # GoalTolerance
    
 xy_goal_tolerance: 0.5
 yaw_goal_tolerance: 1.0
 free_goal_vel: False
    
 # Obstacles
    
 min_obstacle_dist: 1.0
 include_costmap_obstacles: True
 costmap_obstacles_behind_robot_dist: 1.0
 obstacle_poses_affected: 30 # 5 Lowered for performance
 # Moved to own file:
 # costmap_converter_plugin: "costmap_converter::CostmapToPolygonsDBSMCCH" 
 # costmap_converter_plugin: ""
 # costmap_converter_spin_thread: True
 # costmap_converter_rate: 5

 # Optimization
    
 no_inner_iterations: 7 # 5 Lowered for performance
 no_outer_iterations: 6 # 4 Lowered for performance
 optimization_activate: True
 optimization_verbose: False
 penalty_epsilon: 0.1
 weight_max_vel_x: 5
 weight_max_vel_theta: 1
 weight_acc_lim_x: 3
 weight_acc_lim_theta: 1
 weight_kinematics_nh: 1000
 weight_kinematics_forward_drive: 400
 weight_kinematics_turning_radius: 1
 weight_optimaltime: 2
 weight_obstacle: 50
 weight_dynamic_obstacle: 10 # not in use yet
 alternative_time_cost: False # not in use yet

 # Homotopy Class Planner

 enable_homotopy_class_planning: True
 enable_multithreading: True
 simple_exploration: False
 max_number_classes: 6
 roadmap_graph_no_samples: 15
 roadmap_graph_area_width: 5
 h_signature_prescaler: 0.5
 h_signature_threshold: 0.1
 obstacle_keypoint_offset: 0.1
 obstacle_heading_threshold: 0.45
 visualize_hc_graph: False